java/ql/lib/semmle/code/java/frameworks/android/Intent.qll

```diff
@@ -52,3 +53,114 @@ class BundleGetterMethod extends Method, TaintPreservingCallable {
 
   override predicate returnsTaintFrom(int arg) { arg = -1 }
 }
+
+private class IntentBundleFlowSteps extends SummaryModelCsv {
+  override predicate row(string row) {
+    row =
+      [
+        //"namespace;type;subtypes;name;signature;ext;input;output;kind"
+        "android.os;BaseBundle;true;get;(String);;MapValue of Argument[-1];ReturnValue;value",
+        "android.os;BaseBundle;true;getString;(String);;MapValue of Argument[-1];ReturnValue;value",
+        "android.os;BaseBundle;true;getString;(String,String);;MapValue of Argument[-1];ReturnValue;value",
+        "android.os;BaseBundle;true;getString;(String,String);;Argument[1];ReturnValue;value",
+        "android.os;BaseBundle;true;getStringArray;(String);;MapValue of Argument[-1];ReturnValue;value",
+        "android.os;BaseBundle;true;keySet;();;MapKey of Argument[-1];Element of ReturnValue;value",
+        "android.os;BaseBundle;true;putAll;(PersistableBundle);;MapKey of Argument[0];MapKey of Argument[-1];value",
+        "android.os;BaseBundle;true;putAll;(PersistableBundle);;MapValue of Argument[0];MapValue of Argument[-1];value",
+        "android.os;BaseBundle;true;putBoolean;;;Argument[0];MapKey of Argument[-1];value",
+        "android.os;BaseBundle;true;putBooleanArray;;;Argument[0];MapKey of Argument[-1];value",
+        "android.os;BaseBundle;true;putDouble;;;Argument[0];MapKey of Argument[-1];value",
+        "android.os;BaseBundle;true;putDoubleArray;;;Argument[0];MapKey of Argument[-1];value",
+        "android.os;BaseBundle;true;putInt;;;Argument[0];MapKey of Argument[-1];value",
+        "android.os;BaseBundle;true;putIntArray;;;Argument[0];MapKey of Argument[-1];value",
+        "android.os;BaseBundle;true;putLong;;;Argument[0];MapKey of Argument[-1];value",
+        "android.os;BaseBundle;true;putLongArray;;;Argument[0];MapKey of Argument[-1];value",
+        "android.os;BaseBundle;true;putString;;;Argument[0];MapKey of Argument[-1];value",
+        "android.os;BaseBundle;true;putString;;;Argument[1];MapValue of Argument[-1];value",
+        "android.os;BaseBundle;true;putStringArray;;;Argument[0];MapKey of Argument[-1];value",
+        "android.os;BaseBundle;true;putStringArray;;;Argument[1];MapValue of Argument[-1];value",
+        "android.os;Bundle;true;getBinder;(String);;MapValue of Argument[-1];ReturnValue;value",
+        "android.os;Bundle;true;getBundle;(String);;MapValue of Argument[-1];ReturnValue;value",
+        "android.os;Bundle;true;getByteArray;(String);;MapValue of Argument[-1];ReturnValue;value",
+        "android.os;Bundle;true;getCharArray;(String);;MapValue of Argument[-1];ReturnValue;value",
+        "android.os;Bundle;true;getCharSequence;(String);;MapValue of Argument[-1];ReturnValue;value",
```

Comment:
Consider the model for `getCharSequence(String,CharSequence)` not returning the default value.

```suggestion
        "android.os;Bundle;true;getCharSequence;;;MapValue of Argument[-1];ReturnValue;value",
```